packages/xod-arduino/platform/program.tpl.cpp

```diff
@@ -39,12 +39,63 @@ namespace xod {
 {{/unless}}
 {{/each}}
 
+// TODO: no constexpr because of Serial::find
+char debuggerProtocolPrefix[] = "+XOD:";
+uint8_t prefixLength = strlen(debuggerProtocolPrefix);
+
 void runTransaction() {
     g_transactionTime = millis();
 
     XOD_TRACE_F("Transaction started, t=");
     XOD_TRACE_LN(g_transactionTime);
 
+    #if defined(XOD_DEBUG)
```

Comment:
I suggest using `#ifdef XOD_DEBUG_SERIAL` and `XOD_DEBUG_SERIAL` instead of `DEBUG_SERIAL`, so this code will automatically work fine in the Simulation mode. I hope :)